### DISCUSSION TITLE
Github Support Community for C++ Developers

### DISCUSSION THREAD

**Discussion by bhawesh2002** [community/community#21450] [Other Feature Feedback, Questions, & Ideas]
<p>A support communnity for C++ developers like me  allways looking foreword for some help from other developers.</p>
<p>Ask quuestions on this topic and share it with other developers.</p>

**Comment by airtower-luna** [comment]
<p>The <a class="hashtag" href="/c/software-development/47">#<span>software-development</span></a> category would be the right place if you’d like to discuss programming issues on this forum. I hope that helps!</p>

**Comment by bhawesh2002** [comment]
<p>Can Anyone helpp me with this c++ code.</p>
<pre><code class="lang-auto">#include &lt;iostream&gt;
#include &lt;string&gt;

using namespace std;

struct test{
    string name;
    int age;
};

int main(){
    cout &lt;&lt; "No of students:\t";
    int entries;
    cin &gt;&gt; entries;
    cout &lt;&lt; "Enter the data" &lt;&lt; endl;
    test * data = new test[entries];
    cout &lt;&lt; "Enter Name: ";
    getline(cin,data[0].name);
    cout &lt;&lt; "Enter Age: ";
    cin &gt;&gt; data[0].age;
    return 0;    
}
</code></pre>
<p>The  code implements a structure named <code>test</code> which holds a string and an int.<br>
The main function declares an array of pointer to test named details with new keyword<br>
{<code>test * data = new test[entries];</code>}<br>
I am trying  to take inputs but faceing these problems:-<br>
I am unable to enter  <code>name</code> which is a string type.<br>
It directly jumps to <code>age</code> input.</p>

**Comment by fire-eggs** [comment, marked as answer]
<p><code>cin</code> and <code>getline</code> don’t play nice together. <code>cin</code> leaves a newline in the input buffer, which causes your <code>getline</code> to fail.</p>
<p>You could just use <code>cin</code> instead of <code>getline</code> to populate <code>name</code> or you could call <code>cin.ignore()</code> just before calling <code>getline</code>.</p>
<p>See <a href="https://stackoverflow.com/questions/18786575/using-getline-in-c" rel="noopener nofollow ugc">here</a>.</p>

**Comment by bhawesh2002** [comment]
<p>Thank you  For your Help</p>
